### ISSUE TITLE
Cannot run daemon as bitcoin user on OS X 10.11

### ISSUE THREAD

**Issue by alexreg** [bitcoin/bitcoin#7845]
I attempt the following in the shell:

```
$ sudo -u bitcoin -g bitcoin debug=1 /opt/local/bin/bitcoind -conf="/opt/local/etc/bitcoin/bitcoin.conf" -datadir="/opt/local/var/lib/bitcoind" -pid="/opt/local/var/run/bitcoind/bitcoind.pid"


************************
EXCEPTION: N5boost10filesystem16filesystem_errorE       
boost::filesystem::status: Permission denied: "/Users/alex/Library/Application Support"       
bitcoin in AppInit()       

```

What is going on here? I am not trying to do anything under `/Users/alex/Library/Application Support`. On the contrary, I am running `bitcoind` as a specialised user/group for the system. Could someone please explain? Maybe there is a path I forget to change?

**Comment by alexreg** [comment]
Okay, after some debugging I've tracked this down to the call to `TryCreateDirectory` in `GetDefaultDataDir` in `src/util.cpp`. It's not actually clear why for OS X (and just OS X) a function to return a default path should actually try to create that path – surely that should be done elsewhere and only if necessary to put files within that dir. Anyway, this seems to be the problem... though I can't guarantee it's the only point of failure, it's likely to be the main one.

Do you think we can get this fixed promptly and a new release out, or at least a Git tag? Cheers.

**Comment by kirkalx** [comment]
Strange, seems to me you are right. Try a PR removing that line and see what happens.
The other part of the mystery is due to whatever behaviour sudo has on your version of OSX - presumably it is leaving $HOME set to "/Users/alex".

**Comment by alexreg** [comment]
Indeed. I've given that a try. Looks to be passing, fingers crossed...

**Comment by alexreg** [comment]
All good. Patch is working great for me here, localy.

**Comment by laanwj** [comment]
Solved by #7850
